File: Python/Leetcode 1137.py
class Solution:
    def tribonacci(self, n: int) -> int:
        tribonacii = list()

        tribonacii.append(0)
        tribonacii.append(1)
        tribonacii.append(1)

        if n < 3:
            return tribonacii[n]

        for num in range(2,n):
            tribonacii.append(tribonacii[num] + tribonacii[num-1] + tribonacii[num-2])

        return tribonacii[n]

File: C++/Leetcode 1137.cpp
#include <stdio.h>
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int tribonacci(int n) {
        if (n <= 2)
        {
            return min(1,n);
        }

        int prev3 = 0;
        int prev2 = 1;
        int prev1 = 1;

        int answer = -1;
        for (int i = 3; i <= n; i++)
        {
            answer = prev3 + prev2 + prev1;
            prev3 = prev2;
            prev2 = prev1;
            prev1 = answer;
            // cout << i   << " " <<  answer << endl;
        }
        return answer;
    }
};